tests/integration-tests/tests/storage/test_ebs.py

```diff
@@ -113,6 +113,40 @@ def test_ebs_single_empty(scheduler, pcluster_config_reader, clusters_factory):
     _test_ebs_correctly_shared(remote_command_executor, mount_dir, scheduler_commands)
 
 
+@pytest.mark.dimensions("ap-northeast-2", "c5.xlarge", "alinux2", "sge")
+@pytest.mark.usefixtures("os", "instance")
+def test_ebs_existing(
+    request, vpc_stacks, region, scheduler, pcluster_config_reader, clusters_factory, snapshots_factory
+):
+    logging.info("Testing ebs existing")
+    existing_mount_dir = "existing_mount_dir"
+
+    logging.info("Creating volume")
+
+    volume_id = snapshots_factory.create_existing_volume(
+        request, vpc_stacks[region].cfn_outputs["PublicSubnetId"], region
+    )
+
+    logging.info("Existing Volume id: %s" % volume_id)
+    cluster_config = pcluster_config_reader(
+        volume_id=volume_id,
+        existing_mount_dir=existing_mount_dir,
+    )
+
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    existing_mount_dir = "/" + existing_mount_dir
+    _test_ebs_correctly_mounted(remote_command_executor, existing_mount_dir, volume_size="9.8")
+    _test_ebs_correctly_shared(remote_command_executor, existing_mount_dir, scheduler_commands)
+    # Checks for test data
+    result = remote_command_executor.run_remote_command("cat {}/test.txt".format(existing_mount_dir))
+    assert_that(result.stdout.strip()).is_equal_to("hello world")
+
+    # delete the cluster
+    cluster.delete()
```

Comment:
why delete the cluster here?

Reply:
i think the reason for deleting it is to ensure that the existing volume used when creating the cluster persists through deletion. if so, can you add a check after deleting the cluster that the volume still exists?

Reply:
The reason why we delete the cluster is because we have the fixture to clean up the existing ebs volume. The volume can not be detached if the cluster is still running.

Reply:
Can you still add the check to ensure the pre-existing volume still exists after deleting the cluster? I think it's a good check to have in the test.

Reply:
yes! it is added